app/commands/git/generate_diff_between_exercise_versions.rb

```diff
@@ -22,7 +20,21 @@ def call
     def changed_files
       # This is a diff of two commits considering all files in the respective old and new directories
       raw_diff = `cd #{repo.send(:repo_dir)} && git diff #{old_sha} #{exercise.git_sha} -- #{old_git.dir} #{new_git.dir}` # rubocop:disable Layout/LineLength
-      ProcessDiff.(raw_diff)
+      changes = ProcessDiff.(raw_diff)
+
+      # We now need to check for any new files that have been added to they interesting_paths,
+      # but may have been added to git in a previous commit. In these cases we need to do a manual
+      # diff for each file between the very first commit in the repo, and this latest commit
+      return changes unless new_interesting_paths.present?
```

Comment:
The rubocop fix is for this line. It wants to change unless/present? to if/blank? but I think that just reads worse. I don't believe there's a global rule for this as if/unless have different semantic meanings to the reader. So I've turned the rule off.